File: c++/1644.cpp
#include <iostream>
#include <vector>

using namespace std;

bool isNotPrime[4000001];

int main() {
  freopen("example.txt", "r", stdin);
  ios_base::sync_with_stdio(0);
  cin.tie(0);

  int N;
  cin >> N;

  for (int i = 2; i * i <= N; i++) {
    if (!isNotPrime[i]) {
      for (int j = i * i; j <= N; j += i) {
        isNotPrime[j] = true;
      }
    }
  }
  vector<int> v;

  for (int i = 2; i <= N; i++) {
    if (!isNotPrime[i]) v.push_back(i);
  }

  int p1 = 0;
  int p2 = 0;
  int sum = 2;
  int answer = 0;

  while (p1 <= p2 && p2 < v.size()) {
    if (sum >= N) {
      if (sum == N) answer++;
      sum -= v[p1++];
    } else {
      sum += v[++p2];
    }
  }

  cout << answer;

  return 0;
}

File: js/1644.js
const fs = require('fs');

const N = +fs.readFileSync('example.txt').toString();

const isPrime = Array(N + 1).fill(true);
const pn = [];

for (let i = 2; i * i <= N; i++) {
  if (isPrime[i]) {
    for (let j = i * i; j <= N; j += i) {
      isPrime[j] = false;
    }
  }
}

for (let i = 2; i <= N; i++) {
  if (isPrime[i]) pn.push(i);
}

let p1 = 0;
let p2 = 0;
let sum = 2;
let answer = 0;

while (p2 <= p1 && p1 < pn.length) {
  if (sum >= N) {
    if (sum === N) answer++;
    sum -= pn[p2++];
  } else {
    sum += pn[++p1];
  }
}

console.log(answer);
